protocol-units/bridge/e2e/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use anyhow::Result;
+//use bridge_shared::types::BridgeTransferId;
+//use std::process::Command;
+use hex::encode;
+use std::{env, process::Stdio};
+use tokio::process::Command as TokioCommand;
+use tokio::time::{sleep, Duration};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+	let config_path = "path/to/config.json";
+	// 1st Anvil test address
+	let initiator_address = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
+	let initiator_priv_key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+	let recipient_address = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92267"; // dummy val, this should be a movement address
+	let rpc_url = "http://localhost:8545";
+	let weth_path = "protocol-units/bridge/contracts/src/WETH9.sol:WETH9";
+
+	let hash_lock = "forty-two".as_bytes();
+	let hash_lock_bytes = keccak_hash::keccak(hash_lock);
+	let hash_lock = encode(hash_lock_bytes);
+
+	let time_lock: u64 = 3600; // Example value
+	let amount: u64 = 1000; // Example value
+	let bridge_transfer_id = "bridge_transfer_id";
+	let pre_image = "pre_image";
+
+	let current_dir = env::current_dir()?;
+	println!("Current dir: {:?}", current_dir);
+
+	// Build contracts
+	let build_output = TokioCommand::new("forge")
+		.args(&["build"])
+		.current_dir("protocol-units/bridge/contracts") //navigate to contracts dir
+		.output()
+		.await?;
+	if !build_output.status.success() {
+		eprint!("Failed to build contracts: {}", String::from_utf8_lossy(&build_output.stderr));
+		return Err(anyhow::anyhow!("Failed to build contracts"));
+	} else {
+		println!("forge build output:");
+		println!("{}", String::from_utf8_lossy(&build_output.stdout));
+	}
+
+	//Start Anvil
+	let _ = TokioCommand::new("anvil").stdout(Stdio::null()).stderr(Stdio::null()).spawn()?;
```

Comment:
This probably conceals a fair bit of useful debugging info, no?

Reply:
Yeah good point, I'll update.